lib/mix/lib/mix/tasks/compile.erlang.ex

```diff
@@ -84,11 +92,10 @@ defmodule Mix.Tasks.Compile.Erlang do
 
     compile_path = Path.relative_to(compile_path, File.cwd!())
 
-    tuples =
-      files
-      |> scan_sources(include_path, source_paths)
-      |> sort_dependencies()
-      |> Enum.map(&annotate_target(&1, compile_path, opts[:force]))
+    erls = scan_sources(files, include_path, source_paths)
+    tuples = Enum.map(erls, &annotate_target(&1, compile_path, opts[:force]))
```

Comment:
Shouldn't we do the target annotation in parallel too? (i.e. inside scan_sources).